File: DsaWithCpp/Tree/BinaryTree/CountNodes.cpp
#include <iostream>
#include "Node.cpp"
#include "TreeBuilder.cpp"
using namespace std;

static int countNode(Node *root)
{
    if (root == NULL)
        return 0;
    return countNode(root->left) + countNode(root->right) + 1;
}

// int main()
// {
//     int nodes[] = {1, 2, 4, -1, -1, 5, -1, -1, 3, -1, 6, -1, -1};
//     Node *root = buildTree(nodes);
//     cout << countNode(root) << endl;
// }

File: Python/tree/binary_tree/count_nodes.py
from node import Node
from tree_builder import build_tree


def count_node(root: Node) -> int:
    if root is None:
        return 0
    return count_node(root.left) + count_node(root.right) + 1


def main():
    nodes = [1, 2, 4, -1, -1, 5, -1, -1, 3, -1, 6, -1, -1]
    root = build_tree(nodes)
    print(count_node(root))


if __name__ == "__main__":
    main()
